index.html

```diff
@@ -140,67 +166,250 @@ <h2>
       </p>
       <p>
         When presenting a badge, it SHOULD be formatted according to the user's
-        <a data-cite="ltli/#locale">locale</a> settings.
-
-        For example, the badge content '7' should be displayed as '7' in the
-        <a data-cite="ltli/#locale">locale</a> 'en-NZ' but as '٧' in 'ar-EG'.
-      </p>
-      <p class="note">
-        The user agent can choose to clear the badge if it deems it
-        necessary, such as when site data is cleared.
+        <a data-cite="ltli/#locale">locale</a> settings. For example, the badge
+        content '7' should be displayed as '7' in the <a data-cite=
+        "ltli/#locale">locale</a> 'en-NZ' but as '٧' in 'ar-EG'.
       </p>
     </section>
-    <section data-dfn-for="Badge">
+    <section>
       <h2>
-        <code>Badge</code> namespace
+        `Navigator` and `WorkerNavigator` interfaces
       </h2>
+      <p>
+        The {{Navigator}} and {{WorkerNavigator}} interfaces are extended with
+        methods for setting and clearing both the document and application
+        badge indicators from documents and service workers, respectively.
+      </p>
       <pre class="idl">
-        [Exposed=(Window)]
-        namespace Badge {
-          void set(optional [EnforceRange] unsigned long long contents);
-          void clear();
+        // Methods only exposed on documents.
+        [SecureContext]
+        partial interface Navigator {
+          Promise&lt;void&gt; setClientBadge(optional [EnforceRange] unsigned long long contents);
+          Promise&lt;void&gt; clearClientBadge();
+        };
+
+        // Methods exposed on both documents and service workers.
+        [SecureContext]
+        interface mixin NavigatorBadge {
+          Promise&lt;void&gt; setAppBadge(optional [EnforceRange] unsigned long long contents);
```

Comment:
Out of curiosity, could `contents` be an
```webidl
optional [EnforceRange] unsigned long long? contents
```

I think with our current definition navigator.setAppBadge(undefined) ==> navigator.setAppBadge() ==> Set badge to flag. Is this desirable?

Reply:
(navigator.setAppBadge(null) ==> navigator.clearAppBadge())